packages/core/src/components/modal/test/open-close/modal.e2e.ts

```diff
@@ -0,0 +1,55 @@
+import { test } from 'stencil-playwright';
+import { expect } from '@playwright/test';
+
+const componentTestPath = 'src/components/modal/test/open-close/index.html';
+
+test.describe('tds-modal-open', () => {
+  test.beforeEach(async ({ page }) => {
+    await page.goto(componentTestPath);
+  });
+
+  test('1. Modal is opened by default', async ({ page }) => {
+    const tdsModal = page.locator('tds-modal');
+    const hasShowAttribute = await tdsModal.getAttribute('show');
+    if (hasShowAttribute) {
+      await expect(tdsModal).toBeVisible(); // Modal is initially open
+      await expect(page).toHaveScreenshot({ maxDiffPixels: 0 });
+    }
+  });
+
+  test('Clicking [Delete] button closes the modal', async ({ page }) => {
+    const deleteButton = page.getByText('Delete');
+    await deleteButton.click(); // Click [Delete] button
+    const tdsModal = page.locator('tds-modal');
```

Comment:
You can also see if the button that you just clicked is still visible or not I think.


```
    const deleteButton = page.getByRole('button').filter({ hasText: /Delete/ });
    await expect(deleteButton).toBeVisible(); // button to be visible
    await deleteButton.click(); // Click [Delete] button
    await expect(deleteButton).toBeHidden(); // button is removed
```